services/workflows-service/src/customer/customer.controller.external.intg.test.ts

```diff
@@ -0,0 +1,205 @@
+import { Customer, Project } from '@prisma/client';
+import { createProject } from '@/test/helpers/create-project';
+import { UserService } from '@/user/user.service';
+import { SalesforceService } from '@/salesforce/salesforce.service';
+import { SalesforceIntegrationRepository } from '@/salesforce/salesforce-integration.repository';
+import { UserRepository } from '@/user/user.repository';
+import { PasswordService } from '@/auth/password/password.service';
+import { WorkflowTokenService } from '@/auth/workflow-token/workflow-token.service';
+import { WorkflowTokenRepository } from '@/auth/workflow-token/workflow-token.repository';
+import { ClsModule } from 'nestjs-cls';
+import { UiDefinitionService } from '@/ui-definition/ui-definition.service';
+import { UiDefinitionRepository } from '@/ui-definition/ui-definition.repository';
+import { CustomerService } from './customer.service';
+import { CustomerControllerExternal } from './customer.controller.external';
+import { CustomerRepository } from './customer.repository';
+import { EndUserService } from '@/end-user/end-user.service';
+import { before } from 'lodash';
+import { AllExceptionsFilter } from '@/common/filters/AllExceptions.filter';
+import { string } from 'zod';
+
+describe('#CustomerControllerExternal', () => {
+  let app: INestApplication;
+  let customerService: CustomerService;
+  let project: Project;
+  let customer: Customer;
+  beforeAll(cleanupDatabase);
+  afterEach(tearDownDatabase);
+
+  beforeAll(async () => {
+    // await cleanupDatabase();
+
+    const servicesProviders = [
+      AllExceptionsFilter,
+      CustomerService,
+      CustomerRepository,
+      EndUserService,
+      EndUserRepository,
+      EntityRepository,
+      FilterService,
+      ProjectScopeService,
+      FilterRepository,
+      FileRepository,
+      FileService,
+      StorageService,
+      WorkflowEventEmitterService,
+      BusinessRepository,
+      WorkflowDefinitionRepository,
+      WorkflowRuntimeDataRepository,
+      WorkflowService,
+      EventEmitter2,
+      PrismaService,
+      UserService,
+      UserRepository,
+      SalesforceService,
+      SalesforceIntegrationRepository,
+      PasswordService,
+      WorkflowTokenService,
+      WorkflowTokenRepository,
+      WorkflowRuntimeDataRepository,
+      UiDefinitionRepository,
+      UiDefinitionService,
+    ];
+    customerService = (await fetchServiceFromModule(CustomerService, servicesProviders, [
+      PrismaModule,
+    ])) as unknown as CustomerService;
+    app = await initiateNestApp(
+      app,
+      [
+        {
+          provide: CustomerService,
+          useValue: customerService,
+        },
+        ...servicesProviders,
+      ],
+      [CustomerControllerExternal],
+      [PrismaModule, ClsModule],
+    );
+  });
+
+  describe('POST /subscriptions', () => {
+    let customerId: string;
+    let projectId: string;
+
+    beforeEach(async () => {
```

Comment:
Why is there an `expect` in a `beforeEach` and the `beforeEach` is in a `describe`?

Reply:
to make sure the client which should add customer did what he actually should do